lib/assets/javascripts/turbograft/turbolinks.coffee

```diff
@@ -56,6 +56,68 @@ removeNode = (node) ->
   removedNode = node.parentNode.removeChild(node)
   triggerEvent('page:after-node-removed', removedNode)
 
+updateHead = (doc) ->
+  loadedAssets = extractTrackAssets document
+  fetchedAssets = extractTrackAssets doc
+
+  updateLinkTags(loadedAssets, fetchedAssets)
+  updateScriptTags(loadedAssets, fetchedAssets)
+
+extractTrackAssets = (doc) ->
+  for node in doc.querySelector('head').childNodes when node.getAttribute?('data-turbolinks-track')?
+    node
+
+filterForNodeType = (nodeList, nodeType) ->
+  (node for node in nodeList when node.nodeName == nodeType)
```

Comment:
I super prefer filters and maps. I was squeemish about the file's conventions and browser support and whatnot, but I'm leaning more towards just using the good things anyway.